### PR TITLE
Processor tests

### PR DESCRIPTION
**Pull request by rettigl**
This PR implements tests for the processor methods, intends to clean up the processor code somewhat, and fix bugs detected on the way

### CODE REVIEW

**Review by rettigl — COMMENTED** [review]
This PR is ready for review from my side now. It has grown quite big again, but I discovered a number of inconsistencies and small bugs while writing these tests, which I have fixed along the way. I provided a summary of the changes in each file to make review easiert.

File: tests/calibrator/test_momentum.py
[the remark this answers is not in the document]
Comment (rettigl) [reply tests/calibrator/test_momentum.py:1]: Also here, the mpes_example config has been removed now.

File: tests/calibrator/test_delay.py
Comment (rettigl) [inline tests/calibrator/test_delay.py:1]: Also here, the mpes_example config has been removed now.

**Review by zain-sohail — APPROVED** [review]
Looked through the changelog you provided and that is very helpful! 
I have reviewed the PR, not with very fine detail but I think since it's mostly test additions which are passing, and some other minor fixes I think it is good to merge.

**Comment by rettigl** [comment]
Closes #140, closes #141, closes #153

**Review by steinnymir — COMMENTED** [review]
Change log was a great idea. I looked into the parts where I could have a say and all looks good to me!

**Pull request merged**